pkg/appgw/frontend_listeners.go

```diff
@@ -194,12 +207,18 @@ func (c *appGwConfigBuilder) groupListenersByListenerIdentifier(cbCtx *ConfigBui
 	for idx, listener := range *listeners {
 		port, portExists := portsByID[*listener.FrontendPort.ID]
 
-		listenerID := listenerIdentifier{
+		listenerID := listenerIdentifier {
 			UsePrivateIP: IsPrivateIPConfiguration(LookupIPConfigurationByID(c.appGw.FrontendIPConfigurations, listener.FrontendIPConfiguration.ID)),
 		}
-		if listener.HostName != nil {
-			listenerID.HostName = *listener.HostName
+
+		if listener.Hostnames != nil {
+			var hostnames []string = *listener.Hostnames
+			listenerID.setHostNames(hostnames)
```

Comment:
You could combine the 2 lines into one without losing much readability: `listenerID.setHostNames(*listener.Hostnames)`